sisl/viz/plotly/plots/tests/get_files.py

```diff
@@ -0,0 +1,8 @@
+from pathlib import Path
+
+files_root = Path("/home/pfebrer/webDevelopement/sislGUI/sisl/sisl/viz/plotly/Notebooks/files")
```

Comment:
I think this file shouldn't be there? What did you intent?

Reply:
This is just because at the beggining I didn't know how sisl-files worked so I was using my local files to run the test. But maybe now I could figure it out. There's a fixture called `sisl_files` right?

Reply:
Yes, `sisl_files` fixture is the way to go. This just requires the test-files you need to be in the https://github.com/zerothi/sisl-files repo.

Reply:
And what's the best way to get the files in the tutorial notebooks?

Reply:
In the tutorials it will be a bit more problematic. I would suggest we do a minimal tight-binding, possibly with multiple orbitals so that it is easy to follow and understand. While I agree that using full Siesta basis it isn't really necessary.

In my other tutorials I start from basic tight-binding and show all from the same principle. I think this is the way forward.
Could you have a go at this?

Reply:
The problem is that in this case I would say it's a key feature that the output files can be directly plotted, in the sense that after you run a simulation you can just go `Plot(file)` or `sisl.get_sile(file).plot()`. In my opinion this is where the viz framework adds more value.

Of course you can also use it from a tight-binding hamiltonian (e.g. `H.plot.pdos()`) or a `BandStructure` (e.g. `band_struct.plot.fatbands()`), but people who code their own tight-binding are probably less needed of the plotting framework (although I'm sure it can really help that the visualizations are already coded).

So I would say that the notebooks/tutorials would have less of an impact if I don't use output files. What do you think?

Reply:
> The problem is that in this case I would say it's a key feature that the output files can be directly plotted, in the sense that after you run a simulation you can just go `Plot(file)` or `sisl.get_sile(file).plot()`. In my opinion this is where the viz framework adds more value.

Agreed, probably what could be done is splitting the tutorial in 2, the first part creates the electronic structure and writes the files that will be plotted in the subsequent part. This will also show that the viz module is not limited to siesta output, but could interact with any part of the the siles available?

> Of course you can also use it from a tight-binding hamiltonian (e.g. `H.plot.pdos()`) or a `BandStructure` (e.g. `band_struct.plot.fatbands()`), but people who code their own tight-binding are probably less needed of the plotting framework (although I'm sure it can really help that the visualizations are already coded).
> 
> So I would say that the notebooks/tutorials would have less of an impact if I don't use output files. What do you think?

I think the combination of both would be ideal, writing DM/HSX/TSHS files would be extremely useful and also give clearance for the path of the viz module.
What do you think about this?

I am just very reluctant to cluttering the repo with many binary files, the repo size blows up and I would rather be hard on this than allow one user to add files, and then say no to the next 10....

Reply:
> Agreed, probably what could be done is splitting the tutorial in 2, the first part creates the electronic structure and writes the files that will be plotted in the subsequent part.

I think this useful to display all capabilities of sisl but my idea for the showcase notebooks (i.e. these ones
![image](https://user-images.githubusercontent.com/42074085/93076986-7ceb4500-f688-11ea-8ab6-c1f1a4ed940f.png)
), was to focus deeply on the capabilities of each particular plot, not to provide a full tutorial of how to do things. The main point of this is that you can use it as a reference when you don't know how to manipulate the plot (kind of like a visual and more extended docstring). Then, the fact that you have these notebooks that are really only documentation allows you to create tutorials that don't need to explain the steps where you manipulate the plot, since you can always refer to them. Therefore I am afraid that adding other things to the showcase notebook would blur their aim and make the documentation less clear. 

Having said that, I was thinking if it was worth it to add an "entry points" section that would display what are the different ways of getting that particular plot, but always keeping in mind that is just for the plot class documentation. In the end, I'd like the tree for these showcase notebooks to look like it's a reference page, not a tutorial:

![image](https://user-images.githubusercontent.com/42074085/93078578-056ae500-f68b-11ea-96b1-982c7d971ee9.png)

So what I'm afraid of mostly is that creating a tight-binding model at the beggining of each notebook would grab unnecessary attention. Also, the graphene tight-binding does not have enough features to display the full capabilities of `PdosPlot`, for example (only one species with one orbital).

Reply:
So I agree with the fact that having big files stored is not ideal. Maybe the solution could be to have "preset" dummy tight bindings that can be directly imported just for tests and displaying. I noticed that you have that already for tests, but I would make it an importable thing. 

Most data processing/visualization frameworks have that and it is great to let people investigate the framework without needing to worry about the previous steps. Something like:

```
from sisl.test.hamiltonians import SrTiO3_H
# or
from sisl.test.tight_bindings import graphene_TB
```

and then you can play with it.

But I'm thinking now that, for example, how can I end up with a grid to display `GridPlot` from a hamiltonian in sisl? (there's a specific `WavefunctionPlot` for wavefunctions).

I can probably use files that you already have in `sisl-files` so that we reduce the number of files to the minimum.

Reply:
Agreed, you have very valid points, and I agree.

Your idea about having example tight-binding methods would be good. I like this. Hmm... where should we put this...

As for the `GridPlot`, that is easy ;)
```python
grid = Grid(0.1, geometry=H.geometry, dtype=np.complex128)
H.eigenstate(k=...).sub(2).wavefunction(grid)
```
and you have an eigenstate plotted on the grid. :)

Reply:
> H.eigenstate(k=...).sub(2).wavefunction(grid) 

This is precisely what `WavefunctionPlot` does, that's why I wanted a different example :) But you made me realize that I can just generate a random grid.